CRM/Event/Form/Registration/Confirm.php

```diff
@@ -1065,11 +1068,12 @@ public static function fixLocationFields(&$params, &$fields, &$form) {
       }
     }
 
-    // also add location name to the array
-    if ($form->_values['event']['is_monetary']) {
+    // Add the billing names to the billing address, if a billing name is set
+    if ($form->_values['event']['is_monetary'] && !empty($params['billing_first_name'])) {
```

Comment:
I'd really like to remove the first part of this if - on the basis the decision to record a billing_name should be based on whether we collected it - not a field that probably should not exist anyway.

I'd settle for a code comment